src/components/elements/element_treemap_box.gd

```diff
@@ -8,24 +8,40 @@ var value := 0.0
 var final_size := Vector2.ZERO
 
 onready var text_label = $Center/C/Z/Label
+onready var scaler = $Center/C/Z
 
 
 func _ready():
 	text_label.text = label + "\n(" + str(value) + ")"
 	$Button.hint_tooltip = label + ": " + str(value)
 	$Button.modulate = element_color
+	var font_color = Style.col_map[Style.c.BG2] if element_color.get_luminance() > 0.46 else Style.col_map[Style.c.LIGHT]
+	text_label.add_color_override("font_color", font_color)
 	yield(VisualServer, "frame_post_draw")
 	update_label_pos()
 
 
 func update_label_pos():
-	if final_size.x < 100 or final_size.y < 80:
+	text_label.show()
+	if final_size.x < 30 or final_size.y < 30:
 		text_label.hide()
+	elif final_size.x < 60 or final_size.y < 40:
+		text_label.rect_size.x = final_size.x - 2
+		text_label.rect_size.y = final_size.y - 2
+		text_label.rect_size /= 0.5
```

Comment:
Is there a reason why we dont use `*= 2` here instead?

Reply:
Yes and no. I used 0.5 because I use the same value to scale the `scaler`:
`scaler.scale = Vector2.ONE*0.5`

I thought about calculating the value automatically, so I wanted to be able to use the same value.
But yes, I can change it to cheaper multiplication :)